checkov/sast/runner.py

```diff
@@ -0,0 +1,147 @@
+        output_handler = OutputHandler(output_settings)
+
+        registry.load_checks(runner_filter.sast_languages)
+        if external_checks_dir:
+            for external_checks in external_checks_dir:
+                registry.load_external_checks(external_checks, runner_filter.sast_languages)
+
+        if root_folder:
+            targets = [root_folder]
+        if files:
+            targets = files
+        config = registry.checks
+        if not config:
+            logger.warning('no valid checks')
+            return Report(self.check_type)
+
+        semgrep_output = Runner._get_semgrep_output(targets=targets, config=config, output_handler=output_handler)
+        report = self._get_report(semgrep_output.matches)
+        return report
+
+    @staticmethod
+    def _get_semgrep_output(targets: List[str], config: List[str], output_handler: OutputHandler) -> SemgrepOutput:
+        (filtered_matches_by_rule,
+         semgrep_errors,
+         all_targets,
+         renamed_targets,
+         target_manager_ignore_log,
+         filtered_rules,
+         profiler,
+         profiling_data,
+         parsing_data,
+         explanations,
+         shown_severities,
+         target_manager_lockfile_scan_info) = run_semgrep(output_handler=output_handler, target=targets,
+                                                          pattern="", lang="", configs=config, **{})
+        semgrep_output = SemgrepOutput(filtered_matches_by_rule, semgrep_errors, all_targets, renamed_targets,
+                                       target_manager_ignore_log, filtered_rules, profiler, profiling_data,
+                                       parsing_data, explanations, shown_severities, target_manager_lockfile_scan_info)
+        return semgrep_output
+
+    def _get_report(self, filtered_matches_by_rule: Dict[Rule, List[RuleMatch]]) -> Report:
+        report = Report(self.check_type)
+        for rule, matches in filtered_matches_by_rule.items():
+            for match in matches:
+                check_id = rule.id.split('.')[-1]
+                check_name = rule.metadata.get('name', '')
+                code_block = Runner._get_code_block(match.lines, match.start.line)
+                file_abs_path = match.match.location.path
+                file_path = file_abs_path.split('/')[-1]
+                severity = get_severity(SEMGREP_SEVERITY_TO_CHECKOV_SEVERITY.get(rule.severity))
+                file_line_range = [match.start.line, match.end.line]
+                check_result = {'result': CheckResult.FAILED, 'evaluated_keys': {}}
+
+                record = Record(check_id=check_id, check_name=check_name, resource=None, evaluations={},
+                                check_class=None, check_result=check_result, code_block=code_block,
+                                file_path=file_path, file_line_range=file_line_range,
+                                file_abs_path=file_abs_path, severity=severity)
```

Comment:
```suggestion
                check_result = _CheckResult(result=CheckResult.FAILED)

                record = Record(check_id=check_id, check_name=check_name, resource="", evaluations={},
                                check_class="", check_result=check_result, code_block=code_block,
                                file_path=file_path, file_line_range=file_line_range,
                                file_abs_path=file_abs_path, severity=severity)
```

Had some typing mismatches